core/config/value.go

```diff
@@ -487,8 +486,26 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, boo
 			if destSlice.Len() > 0 {
 				fieldValue.Set(destSlice)
 			}
-		}
+		case bucketMap:
+			createMap := reflect.MakeMap(fieldType).Interface()
+			val := global.GetValue(childKey).Value()
 
+			if val != nil {
+				switch v := val.(type) {
+				case map[interface{}]interface{}:
+					destMap := createMap.(map[string]interface{})
+					for key, value := range v {
+						switch key := key.(type) {
+						case string:
+							destMap[key] = value
+						default:
+							panic(fmt.Sprintf("Can't parse the input yaml. Map key must be string: %v", key))
```

Comment:
is panic() the right option here? Is there another way we can bubble up the error? Under what circumstances would we encounter this path?